### PR TITLE
Increase mon failover timeout to 15 min for IBMCloudPlatformType

### PR DESCRIPTION
**Pull request by rohan47**
mon C take time to come up on IBMCloudPlatform which timeouts and Mon d gets created. So increase the mon
failover timeout to 15 min for IBMCloudPlatformType to allow enough time
for mon c to come up.

Signed-off-by: rohan47 <rohgupta@redhat.com>

### CODE REVIEW

**Comment by rohan47** [comment]
not yet tested on an actual cluster
/hold

**Comment by rohan47** [comment]
Tested the code changes.
/hold

File: controllers/storagecluster/cephcluster.go
Comment (jarrpa) [inline controllers/storagecluster/cephcluster.go:120]: There is a new const being introduced in another IBM PR: https://github.com/openshift/ocs-operator/blob/8c0b66ade6c1187c68cdc8334ee7a9331272b45b/controllers/storagecluster/platform_detection.go#L21 (PR: https://github.com/openshift/ocs-operator/pull/1088)

We should probably wait until that one merges and add the new PlatformType in your if statement.

File: controllers/storagecluster/cephcluster_test.go
Comment (jarrpa) [inline controllers/storagecluster/cephcluster_test.go:106]: Since this test is not required to pass the unit tests of the code changes, add the test in a separate commit.

**Review by jarrpa — CHANGES_REQUESTED** [review]

File: controllers/storagecluster/cephcluster_test.go
Comment (rohan47) [reply controllers/storagecluster/cephcluster_test.go:106]: Done

File: controllers/storagecluster/cephcluster.go
Comment (rohan47) [reply controllers/storagecluster/cephcluster.go]: Added.

File: controllers/storagecluster/cephcluster.go
Comment (jarrpa) [inline controllers/storagecluster/cephcluster.go:120]: Don't reference the variable directly, use the GetPlatform() function.

**Review by jarrpa — CHANGES_REQUESTED** [review]

File: controllers/storagecluster/cephcluster.go
Comment (rohan47) [reply controllers/storagecluster/cephcluster.go]: Done

**Review by jarrpa — APPROVED** [review]
/lgtm

File: controllers/storagecluster/cephcluster.go
Comment (sp98) [inline controllers/storagecluster/cephcluster.go:122]: should be logged as error.

File: controllers/storagecluster/cephcluster_test.go
Comment (sp98) [inline controllers/storagecluster/cephcluster_test.go:113]: its nice to add more details about the `label` rather than just `case 1`.

File: controllers/storagecluster/cephcluster.go
Comment (sp98) [reply controllers/storagecluster/cephcluster.go:122]: shouldn't we return in case of error ?

**Review by sp98 — CHANGES_REQUESTED** [review]

File: controllers/storagecluster/cephcluster.go
Comment (sp98) [inline controllers/storagecluster/cephcluster.go:125]: nit. would be nice to log both the platform name and `900s` timeout.

File: controllers/storagecluster/cephcluster_test.go
Comment (sp98) [inline controllers/storagecluster/cephcluster_test.go:145]: is the print statement needed here? if yes, then we can add some more context to it. 
say, 
`fmt.Println("Platform:", c.platform)`

**Review by sp98 — COMMENTED** [review]
nit

File: controllers/storagecluster/cephcluster.go
Comment (rohan47) [reply controllers/storagecluster/cephcluster.go:122]: We are changing the mon failover timeout for IBM platform only, but if it fails I don't think it should be a problem.
So maybe instead of just logging it, Throwing a warning would be better.

Comment (sp98) [reply controllers/storagecluster/cephcluster.go:122]: ok. so if we don't want to return error here, then warning should be fine and next if block should be skipped.

File: controllers/storagecluster/cephcluster_test.go
Comment (rohan47) [reply controllers/storagecluster/cephcluster_test.go:145]: was using this to debug and forgot to remove.
Its not required

File: controllers/storagecluster/cephcluster.go
Comment (rohan47) [reply controllers/storagecluster/cephcluster.go]: Logging the error

File: controllers/storagecluster/cephcluster.go
Comment (travisn) [inline controllers/storagecluster/cephcluster.go:125]: For readability you could set `15m`, but the result will be the same.
```suggestion
		cephCluster.Spec.HealthCheck.DaemonHealth.Monitor.Timeout = "15m"
```

File: controllers/storagecluster/cephcluster_test.go
Comment (sp98) [inline controllers/storagecluster/cephcluster_test.go:113]:
```suggestion
			label:    "case 1", // when the platform is not identified
```

File: controllers/storagecluster/cephcluster.go
Comment (rohan47) [reply controllers/storagecluster/cephcluster.go]: done

File: controllers/storagecluster/cephcluster_test.go
Comment (rohan47) [reply controllers/storagecluster/cephcluster_test.go]: Done

**Review by jarrpa — APPROVED** [review]
/lgtm

**Comment by agarwal-mudit** [comment]
@rohan47 I think we can remove DNM label now? Its ready to merge, right?

File: controllers/storagecluster/cephcluster.go
Comment (sp98) [inline controllers/storagecluster/cephcluster.go:124]: Log usage is not correct.  Its `Info(msg string, keysAndValues ...interface{})`.  You can check other examples in OCS for reference.

**Review by sp98 — CHANGES_REQUESTED** [review]

File: controllers/storagecluster/cephcluster.go
Comment (rohan47) [reply controllers/storagecluster/cephcluster.go]: Changed

**Review by sp98 — APPROVED** [review]

**Review by umangachapagain — COMMENTED** [review]
/lgtm

**Pull request merged**
